ReactiveXComponent/WebSocket/WebSocketXCApiManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Threading;
+using Newtonsoft.Json;
+using ReactiveXComponent.Common;
+
+namespace ReactiveXComponent.WebSocket
+{
+    public class WebSocketXCApiManager
+    {
+        private readonly IWebSocketClient _webSocketClient;
+
+        private event EventHandler<List<string>> XCApiNamesReceived;
+        private event EventHandler<string> XCApiReceived;
+
+        private readonly IObservable<List<string>> _xcApiNamesStream;
+        private readonly IObservable<string> _xcApiStream;
+
+        public WebSocketXCApiManager(IWebSocketClient webSocketClient)
+        {
+            _webSocketClient = webSocketClient;
+
+            _xcApiNamesStream = Observable.FromEvent<EventHandler<List<string>>, List<string>>(
+                handler => (sender, e) => handler(e),
+                h => XCApiNamesReceived += h,
+                h => XCApiNamesReceived -= h);
+
+            _xcApiStream = Observable.FromEvent<EventHandler<string>, string>(
+                handler => (sender, e) => handler(e),
+                h => XCApiReceived += h,
+                h => XCApiReceived -= h);
+        }
+
+        public List<string> GetXCApiNames(int timeout = 10000)
+        {
+            List<string> result = null;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<List<string>>(message =>
+            {
+                result = new List<string>(message);
+                lockEvent.Set();
+            });
+            var xcApiSubscription = _xcApiNamesStream.Subscribe(observer);
+
+            EventHandler<WebSocketMessageEventArgs> subscriptionHandler;
+            
+            SubscribeXCApiNames(out subscriptionHandler);
+            lockEvent.WaitOne(timeout);
+            
+            xcApiSubscription.Dispose();
+            _webSocketClient.MessageReceived -= subscriptionHandler;
+
+            return result;
+        }
+
+
+        public string GetXCApi(string apiFullName, int timeout = 10000)
+        {
+            var result = string.Empty;
+            var lockEvent = new AutoResetEvent(false);
+            var observer = Observer.Create<string>(message =>
+            {
+                result = message;
+                lockEvent.Set();
+            });
+            var xcApiSubscription = _xcApiStream.Subscribe(observer);
+
+            EventHandler<WebSocketMessageEventArgs> subscriptionHandler;
+
+            SubscribeXCApi(apiFullName, out subscriptionHandler);
```

Comment:
This method creates a handler than subscribes it to an event (__webSocketClient.MessageReceived += subscriptionHandler;_). Then you **manually** unsubscribe that handler (__webSocketClient.MessageReceived -= subscriptionHandler;_). I suggest either getting rid of that method or turning it into a sort of Handler factory (just returns a handler based on the request you send and the response you expect).